modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -1343,7 +1343,12 @@ void WindowContextTop::exit_fullscreen() {
 }
 
 void WindowContextTop::request_focus() {
-    gtk_window_present(GTK_WINDOW(gtk_widget));
+    //JDK-8212060: Window show and then move glitch.
+    //The WindowContextBase::set_visible will take care of showing the window.
+    //The below code will only handle later request_focus.
+    if (is_visible()) {
+        gtk_window_present(GTK_WINDOW(gtk_widget));
```

Comment:
This looks OK as long as you are sure that `gtk_window_present` is unneeded when the window is set visible. Otherwise, it might be safer to set a `presentPending` flag in the `else` case and call `gtk_window_present` in set_visible if `presentPending` is `true`. My suggestion is overkill if the call to `gtk_widget_show all` effectively does a present as well.

Reply:
From gtk docs:

> Presents a window to the user. This may mean raising the window in the stacking order, deiconifying it, moving it to the current desktop, and/or giving it the keyboard focus, possibly dependent on the user’s platform, window manager, and preferences.
> 
> If window is hidden, this function calls gtk_widget_show() as well.

So gtk_widget_show_all is sufficient.